src/fdcan.rs

```diff
@@ -1159,10 +1159,10 @@ where
     #[inline]
     pub fn error_counters(&self) -> ErrorCounters {
         let can = self.registers();
-        let cel: u8 = can.ecr.read().cel().bits();
-        let rp: bool = can.ecr.read().rp().bit();
-        let rec: u8 = can.ecr.read().rec().bits();
-        let tec: u8 = can.ecr.read().tec().bits();
+        let cel: u8 = can.ecr().read().cel().bits();
+        let rp: bool = can.ecr().read().rp().bit();
+        let rec: u8 = can.ecr().read().rec().bits();
+        let tec: u8 = can.ecr().read().tec().bits();
```

Comment:
This do `read` operation 4 times.
```suggestion
        let ecr = can.ecr().read();
        let cel: u8 = ecr.cel().bits();
        let rp: bool = ecr.rp().bit();
        let rec: u8 = ecr.rec().bits();
        let tec: u8 = ecr.tec().bits();
```

Reply:
Oh, nice.

Thanks